spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/types/external.scala

```diff
@@ -182,16 +182,41 @@ object external {
     AtlasEntityWithDependencies(kafkaEntity)
   }
 
+  // ================ RDBMS based entities ======================
+  val RDBMS_TABLE = "rdbms_table"
+
+  /**
+   * Converts JDBC RDBMS properties into Atlas entity.
+   */
+  def rdbmsTableToEntity(url: String, tableName: String): AtlasEntityWithDependencies = {
+    val jdbcEntity = new AtlasEntity(RDBMS_TABLE)
+
+    val databaseName = JdbcUtils.getDatabaseName(url)
+    jdbcEntity.setAttribute("qualifiedName", getRdbmsQualifiedName(databaseName, tableName))
+    jdbcEntity.setAttribute("name", tableName)
+
+    AtlasEntityWithDependencies(jdbcEntity)
+  }
+
+  /**
+   * Constructs the the full qualified name of the database.
+   */
+  private def getRdbmsQualifiedName(databaseName: String, tableName: String): String =
+    s"${databaseName.toLowerCase}.${tableName.toLowerCase}"
+
   // ================== Spark's Hive Catalog entities =====================
 
-  // Note: given that we use Spark model types for Hive catalog entities (except HWC),
-  // Hive catalog entities should follow Spark model definitions.
-  // In Atlas, the attributes which are not in definition are ignored with WARN messages.
+  // Note: SAC always creates Hive table and corresponding entities, whereas SAC refers
+  // existing table entity on HWC.
 
-  val HIVE_DB_TYPE_STRING = metadata.DB_TYPE_STRING
-  val HIVE_STORAGEDESC_TYPE_STRING = metadata.STORAGEDESC_TYPE_STRING
-  val HIVE_TABLE_TYPE_STRING = metadata.TABLE_TYPE_STRING
+  val HIVE_DB_TYPE_STRING = "hive_db"
+  val HIVE_STORAGEDESC_TYPE_STRING = "hive_storagedesc"
+  val HIVE_TABLE_TYPE_STRING = "hive_table"
```

Comment:
Will this work unless the Hive Entity we create from SAC confirms to the model defined by Hive ? I guess we need to keep the SAC logic in sync with Hive models else things can break. (E.g. if hive decides to change their model). Should we rely only on HMS to create these entities since they are the ones maintaining the Hive Atlas model as well?

Reply:
Just commented regarding this.

Reply:
If we can expect Hive to always maintain backward compatibility and the model works like upsert may be OK.